services/dataService.js

```diff
@@ -0,0 +1,62 @@
+const productMap = {};
+const categoryMap = {};
```

Comment:
See above comment. It's editable, so should be `let`.
```suggestion
let categoryMap = null;
```